handle empty value for included listing in vuetify/block_listing

When an .adoc file includes source content delimited by tags and there is no content a freemarker exception is thrown.

An example of an adoc file reproducing this would be like this:

```adoc
// tag::importsHealth3[]
// end::importsHealth3[]
```

The exception is as follow:
```
[ERROR] ----
[ERROR] FTL stack trace ("~" means nesting-related):
[ERROR] 	- Failed at: ${content}  [in template "vuetify/document" at line 21, column 1]
[ERROR] ----: InvocationTargetException: An error occurred during rendering of vuetify/block_section: Error during getter invocation: node=org.asciidoctor.ast.impl.SectionImpl@4a30c120, methodname=getContent
[ERROR] 
[ERROR] ----
[ERROR] FTL stack trace ("~" means nesting-related):
[ERROR] 	- Failed at: #assign _content = content  [in template "vuetify/block_section" at line 16, column 1]
[ERROR] ----: Error during getter invocation: node=org.asciidoctor.ast.impl.SectionImpl@66ce9b99, methodname=getContent
[ERROR] 
[ERROR] ----
[ERROR] FTL stack trace ("~" means nesting-related):
[ERROR] 	- Failed at: #assign _content = content  [in template "vuetify/block_section" at line 16, column 1]
[ERROR] ----: Error during getter invocation: node=org.asciidoctor.ast.impl.SectionImpl@4e2c82a6, methodname=getContent
[ERROR] 
[ERROR] ----
[ERROR] FTL stack trace ("~" means nesting-related):
[ERROR] 	- Failed at: #assign _content = content  [in template "vuetify/block_section" at line 16, column 1]
[ERROR] ----: An error occurred during rendering of vuetify/block_listing: The following has evaluated to null or missing:
[ERROR] ==> content  [in template "vuetify/block_listing" at line 20, column 4]
[ERROR] 
[ERROR] ----
[ERROR] Tip: If the failing expression is known to be legally refer to something that's sometimes null or missing, either specify a default value like myOptionalVar!myDefault, or use <#if myOptionalVar??>when-present<#else>when-missing</#if>. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
[ERROR] ----
[ERROR] 
[ERROR] ----
[ERROR] FTL stack trace ("~" means nesting-related):
[ERROR] 	- Failed at: ${content}  [in template "vuetify/block_listing" at line 20, column 2]
[ERROR] ----
[ERROR] -> [Help 1]
```

The template does expect a non empty value, it should be improved to handle this better: either fail nicely or allow an empty value.

--- a/sitegen/src/main/resources/helidon-sitegen-templates/vuetify/block_listing.ftl
+++ b/sitegen/src/main/resources/helidon-sitegen-templates/vuetify/block_listing.ftl
@@ -17,9 +17,9 @@
 <markup
 <#if attributes["language"]?? >lang="${attributes["language"]}"</#if>
 <#if title??>title="${title}"</#if>
->${content}</markup>
+>${content!""}</markup>
 <#else>
 <div class="listing">
-<pre>${content}</pre>
+<pre>${content!""}</pre>
 </div>
 </#if>

--- a/sitegen/src/main/resources/helidon-sitegen-templates/vuetify/block_listing.ftl
+++ b/sitegen/src/main/resources/helidon-sitegen-templates/vuetify/block_listing.ftl
@@ -13,13 +13,16 @@
   See the License for the specific language governing permissions and
   limitations under the License.
  -->
+<#assign _content = content!"">
+<#if content??>
 <#if style?? && style="source">
 <markup
 <#if attributes["language"]?? >lang="${attributes["language"]}"</#if>
 <#if title??>title="${title}"</#if>
->${content!""}</markup>
+>${_content}</markup>
 <#else>
 <div class="listing">
-<pre>${content!""}</pre>
+<pre>${_content}</pre>
 </div>
+</#if>
 </#if>

--- a/sitegen/src/main/resources/helidon-sitegen-templates/vuetify/block_listing.ftl
+++ b/sitegen/src/main/resources/helidon-sitegen-templates/vuetify/block_listing.ftl
@@ -1,5 +1,5 @@
 <#--
-  Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+  Copyright (c) 2018, 2019 Oracle and/or its affiliates. All rights reserved.
  
   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.

--- a/sitegen/src/test/resources/testvuetify1/IncludeTest.java
+++ b/sitegen/src/test/resources/testvuetify1/IncludeTest.java
@@ -1,0 +1,7 @@
+package testvuetify1;
+
+// tag::emptyInclude[]
+// end::emptyInclude[]
+
+public class IncludeTest {
+}

--- a/sitegen/src/test/resources/testvuetify1/IncludeTest.java
+++ b/sitegen/src/test/resources/testvuetify1/IncludeTest.java
@@ -1,3 +1,18 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package testvuetify1;
 
 // tag::emptyInclude[]

--- a/sitegen/src/test/resources/testvuetify1/about/01_intro.adoc
+++ b/sitegen/src/test/resources/testvuetify1/about/01_intro.adoc
@@ -125,3 +125,10 @@ This is a list:
 == Line through test
 
 Some text in this sentence has some [.line-through]#line through# style.
+
+== Empty include test
+
+[source,java]
+----
+include::../IncludeTest.java[tag=emptyInclude]
+----

